src/components/progress-tracker/progress-tracker.pw.tsx

```diff
@@ -0,0 +1,415 @@
+
+      await checkAccessibility(page);
+    });
+  });
+
+  ([12, 47, 100] as const).forEach((progress) => {
+    test(`should check the accessibility when component is rendered with ${progress}% of progress`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ProgressTrackerComponent progress={progress} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should check the accessibility when component is rendered with error prop", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <ProgressTrackerComponent error showDefaultLabels progress={35} />
+    );
+    await checkAccessibility(page);
+  });
+
+  ([true, false] as const).forEach((boolean) => {
+    test(`should check the accessibility when component is rendered with showDefaultLabels is set to ${boolean}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ProgressTrackerComponent showDefaultLabels={boolean} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  ([CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS] as const).forEach(
+    (currentProgressLabel) => {
+      test(`should check the accessibility when component is rendered with currentProgressLabel is set to ${currentProgressLabel}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <ProgressTrackerComponent
+            currentProgressLabel={currentProgressLabel}
+          />
+        );
+
+        await checkAccessibility(page);
+      });
+    }
+  );
+
+  ([CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS] as const).forEach(
+    (maxProgressLabel) => {
+      test(`should check the accessibility when component is rendered with maxProgressLabel is set to ${maxProgressLabel}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <ProgressTrackerComponent maxProgressLabel={maxProgressLabel} />
+        );
+
+        await checkAccessibility(page);
+      });
+    }
+  );
+
+  ([CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS] as const).forEach(
+    (customValuePreposition) => {
+      test(`should check the accessibility when component is rendered with customValuePreposition is set to ${customValuePreposition}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <ProgressTrackerComponent
+            customValuePreposition={customValuePreposition}
+            showDefaultLabels
+          />
+        );
+
+        await checkAccessibility(page);
+      });
+    }
+  );
+
+  (["top", "bottom"] as ProgressTrackerProps["labelsPosition"][]).forEach(
+    (labelsPosition) => {
+      test(`should check the accessibility when component is rendered with labelsPosition is set to ${labelsPosition}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <ProgressTrackerComponent labelsPosition={labelsPosition} />
+        );
+
+        await checkAccessibility(page);
+      });
+    }
+  );
```

Comment:
**suggestion**: I think for these `labelsPosition` tests, it would be helpful for us to assert if the labels are positioned correctly relative to the tracker line, as currently we are asserting if the text of the labels are correct only.

What do you think about something like this for the "top" test?:

```tsx
const getYValue = (locator: Locator) =>
  locator.evaluate(
    (element) => element.getBoundingClientRect().y
  );

test(`should position current value label below tracker line when labelsPosition prop is "top"`, async ({
  mount,
  page,
}) => {
  await mount(<ProgressTrackerComponent labelsPosition="top" />);

  const currentValueLabel = progressTrackerMinVal(page);
  const trackerLine = progressTrackerLine(page);

  await expect(currentValueLabel).toBeVisible(); // make sure label is visible first
  await expect(trackerLine).toBeVisible(); // make sure tracker line is visible first
  expect(
    await getYValue(currentValueLabel)
  ).toBeLessThan(await getYValue(trackerLine));
});
```